src/main/java/org/jboss/ejb/client/DiscoveryEJBClientInterceptor.java

```diff
@@ -572,7 +573,11 @@ Discovery getDiscovery() {
             if (nodeName == null) {
                 throw Logs.INVOCATION.selectorReturnedNull(selector);
             }
-            location = uris.get(nodeName);
+            if(nodeName.equals(LOCAL_NODE_NAME)){
+                location = Affinity.LOCAL.getUri();
+            } else {
+                location = uris.get(nodeName);
+            }
```

Comment:
Does it work when `exclude-local-receiver` is true (e.g., the user always want invocations to go over network)?

Reply:
no, it will don't respect this setting - I'm working on alternative solution

Reply:
@chengfang  I have created an alternative solution which prioritize local URIs; this solution would respect exclude-local-receiver settings (when it is set LocalEJBReceiver is not registered and only remote URI would be added)